### PR TITLE
fix: get logs from storage

### LINKED ISSUE #1414: Testkube logs should not ask for real pod log in case of status failed or completed
**Describe the bug**

Logs should be taken from pod only if status is "running"
If status is not "running" just get log output from storage

image

### PR DESCRIPTION
**Pull request by vLia**
## Pull request description 
Closes #1414

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: internal/app/api/v1/executions.go [hunk 1]
```diff
@@ -509,6 +501,55 @@ func (s TestkubeAPI) GetExecuteOptions(namespace, id string, request testkube.Ex
 	}, nil
 }
 
+// streamLogsFromStorage writes logs from the output of executionResult to the writer
```
Comment (exu) [inline internal/app/api/v1/executions.go:504]: Why from storage? is there any database used here? As name suggests that it's fetched from some "storage"

File: internal/app/api/v1/executions.go [hunk 2]
```diff
@@ -359,35 +360,26 @@ func (s TestkubeAPI) ExecutionLogsHandler() fiber.Handler {
...
 				return
 			}
 
-			// loop through pods log lines - it's blocking channel
-			// and pass single log output as sse data chunk
-			for out := range logs {
-				s.Log.Debugw("got log", "out", out)
-				fmt.Fprintf(w, "data: ")
-				err = enc.Encode(out)
+			if isExecutionFinished(execution) {
```
Comment (exu) [inline internal/app/api/v1/executions.go:372]: ExecutionResults has this method already

```
func (e *ExecutionResult) IsCompleted() bool {
```

File: internal/app/api/v1/executions.go (hunk 1 above)
Comment (vLia) [reply internal/app/api/v1/executions.go:504]: Storage as opposed to K8s job logs. It makes more sense in the method that it is called from, but there is not direct connection to the storage in this one, only the ExecutionResult that's already queried from there.

would streamFinishedExecutionLogs() be a better name for this?

File: internal/app/api/v1/executions.go (hunk 2 above)
Comment (vLia) [reply internal/app/api/v1/executions.go:372]: As I was testing it with the`test-api-local` tests, I saw that there is 2 more execution status `testkube.SUCCESS_Status` and `testkube.ERROR__Status`, so I checked for those as well in the `isExecutionFinished(...)` function. Tbh I was a bit confused as to why there are 2 different ways to describe the status, but wanted to make sure we catch both

Comment (exu) [reply internal/app/api/v1/executions.go:372]: Holy moly! Good catch - we have file which is not used at all - some leftover after messing with moving status field to 

Please remove this file as it's not used and don't need to be handled anywhere. 

```
pkg/api/v1/testkube/model_status.go
```

And `execution.ExecutionResult.IsCompleted()` will be enough then

File: internal/app/api/v1/executions.go (hunk 1 above)
Comment (exu) [reply internal/app/api/v1/executions.go:504]: It streams execution result if I understand correctly what it does?

Comment (vLia) [reply internal/app/api/v1/executions.go:504]: Yes, it streams the result from the ResultRepository

File: internal/app/api/v1/executions.go (hunk 2 above)
Comment (vLia) [reply internal/app/api/v1/executions.go:372]: Awesome, will do so!

Comment (vLia) [reply internal/app/api/v1/executions.go:372]: Do you know how we can check if there are more files like this?

Comment (exu) [reply internal/app/api/v1/executions.go:372]: probably linter could partially do that - detect unused vars

File: internal/app/api/v1/executions.go (hunk 1 above)
Comment (exu) [reply internal/app/api/v1/executions.go:504]: this function just streams ExecutionResult (not from repository as it's not fetching it) executionResult is passed from outside wherever it'll be so context of this function is not related to repository at all.

Comment (vLia) [reply internal/app/api/v1/executions.go]: fair point, updated it to `streamLogsFromResult(...)`

**Review by exu — APPROVED** [review]

**Pull request merged**